### PR TITLE
Add file question and answer

### LINKED ISSUE #329: Add support for files in forms
(details follow)

### PR DESCRIPTION
**Pull request by open-dynaMIX**
- initial implementation for supporting files
 - depends on MinIO, can easily be extended to other storage providers

Closes #329

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -246,6 +246,31 @@ see [docker-compose.yml](https://github.com/projectcaluma/caluma/blob/master/doc
...
+* `MEDIA_STORAGE_SERVICE`: defaults to "minio" (this is the only supported atm)
+* `MINIO_STORAGE_ENDPOINT`: defaults to "minio:9000"
+* `MINIO_STORAGE_ACCESS_KEY`: defaults to "minio"
+* `MINIO_STORAGE_SECRET_KEY`: defaults to "minio123"
+* `MINIO_STORAGE_USE_HTTPS`: defaults to False
+* `MINIO_STORAGE_MEDIA_BUCKET_NAME`: defaults to "caluma-media"
+* `MINIO_STORAGE_AUTO_CREATE_MEDIA_BUCKET`: defaults to True
+* `MINIO_PRESIGNED_TTL_MINUTES`: defaults to 15
+
+Caluma only handles metadata about files, not the files itself. When saving a `FileAnswer`, Calumna
```
Comment (czosel) [inline README.md]:
```suggestion
Caluma only handles metadata about files, not the files itself. When saving a `FileAnswer`, Caluma
```

File: README.md [hunk 2]
```diff
@@ -246,6 +246,31 @@ see [docker-compose.yml](https://github.com/projectcaluma/caluma/blob/master/doc
...
+will return a presigned `uploadUrl`. So the client can upload the file directly to the storage provider.
```
Comment (czosel) [inline README.md]:
```suggestion
will return a presigned `uploadUrl`, which the client can use to upload the file directly to the storage provider.
```

**Comment by open-dynaMIX** [comment]
@czosel Sorry for the force-push mess. I've accidentally ammended to the wrong commit.

File: caluma/form/serializers.py [hunk 3]
```diff
@@ -607,3 +617,32 @@ def update(self, instance, validated_data):
...
+
+    def update(self, instance, validated_data):
+        if instance.file and instance.file.name is not validated_data["file"]:
+            instance.file.delete()
+            validated_data = self.set_file(validated_data)
+        elif not instance.file:
+            validated_data = self.set_file(validated_data)
+
+        instance = super().update(instance, validated_data)
+        return instance
```
Comment (czosel) [inline caluma/form/serializers.py]: I'd prefer `return super().update(instance, validated_data)`
(also on L635)

Comment (open-dynaMIX) [reply caluma/form/serializers.py]: Thanks for noticing! This was a copy/paste mistake :)

**Comment by czosel** [comment]
@fkm Do you want to try implementing the frontend for this?

**Comment by fkm** [comment]
@czosel I actually need some new task soon. Let's talk about that this afternoon when I come by because of the template service.

File: README.md [hunk 4]
```diff
@@ -246,6 +246,31 @@ see [docker-compose.yml](https://github.com/projectcaluma/caluma/blob/master/doc
...
+* `MINIO_STORAGE_USE_HTTPS`: defaults to False
+* `MINIO_STORAGE_MEDIA_BUCKET_NAME`: defaults to "caluma-media"
+* `MINIO_STORAGE_AUTO_CREATE_MEDIA_BUCKET`: defaults to True
+* `MINIO_PRESIGNED_TTL_MINUTES`: defaults to 15
+
+Caluma only handles metadata about files, not the files itself. When saving a `FileAnswer`, Caluma
+will return a presigned `uploadUrl`, which the client can use to upload the file directly to the storage provider.
+
+The same goes for retrieving files. Caluma will respond with a presigned `downloadUrl` for
+the client to directly download the file from the storage provider.
```
Comment (winged) [inline README.md]: This begs the question, how is file access / security solved?

Comment (open-dynaMIX) [reply README.md]: We're going this route, so caluma never has to deal with actual files. This minimizes overhead from passing files around.

> This begs the question, how is file access / security solved?

I don't know about all the features of MinIO in regard to that. Right now, this is handled by the need to make an authenticated request to caluma in order to get the presigned URL and a low TTL for those URLs.

Comment (winged) [reply README.md]: Ah so the URL contain some cryptographic parts? In this case I think it's OK, and yes the separation of concerns is preferrable

**Comment by czosel** [comment]
:tada:

File: caluma/tests/snapshots/snap_test_schema.py [hunk 5]
```diff
@@ -285,7 +284,7 @@
   form: Form!
   meta: JSONString!
   answers(before: String, after: String, first: Int, last: Int, question: ID, search: String, createdByUser: String, createdByGroup: String, metaHasKey: String, orderBy: [AnswerOrdering]): AnswerConnection
-  parentAnswers: [FormAnswer]
+  parentAnswers: [FileAnswer]
```
Comment (czosel) [inline caluma/tests/snapshots/snap_test_schema.py]: @open-dynaMIX I just noticed this regression when i updated the schema in ember-caluma. Sorry I missed it during review - can you take a look where this is coming from?

**Pull request merged**